src/FSharp.Data.GraphQL.Server.AspNetCore/Giraffe/HttpHandlers.fs

```diff
@@ -115,48 +106,59 @@ module HttpHandlers =
-                                logger.LogTrace(
-                                    $"GraphQL subscription data:\n{{data}}",
-                                    serializeIndented data
-                                )
-                        | SubscriptionErrors(null, errors) ->
-                            logger.LogDebug("Produced GraphQL subscription errors")
+                                logger.LogTrace ($"GraphQL subscription data:\n{{data}}", serializeIndented data)
+                        | SubscriptionErrors (null, errors) ->
+                            logger.LogDebug ("Produced GraphQL subscription errors")
 
                             if logger.IsEnabled LogLevel.Trace then
-                                logger.LogTrace($"GraphQL subscription errors:\n{{errors}}", errors)
-                        | SubscriptionErrors(data, errors) ->
-                            logger.LogDebug("Produced GraphQL subscription result with errors")
+                                logger.LogTrace ($"GraphQL subscription errors:\n{{errors}}", errors)
+                        | SubscriptionErrors (data, errors) ->
+                            logger.LogDebug ("Produced GraphQL subscription result with errors")
 
                             if logger.IsEnabled LogLevel.Trace then
-                                logger.LogTrace(
+                                logger.LogTrace (
                                     $"GraphQL subscription errors:\n{{errors}}\nGraphQL deferred data:\n{{data}}",
                                     errors,
                                     serializeIndented data
                                 ))
 
                 GQLResponse.Stream documentId
             | RequestError errs ->
-                logger.LogWarning(
-                    $"Produced request error GraphQL response with documentId = '{{documentId}}' and metadata:\n{{metadata}}",
-                    documentId,
-                    metadata
-                )
+                if
+                    errs
+                    |> Seq.choose (fun x -> x.Exception |> Option.ofValueOption)
+                    |> (not << Seq.isEmpty)
+                then
+                    errs
+                    |> List.choose (fun x -> x.Exception |> Option.ofValueOption)
+                    |> List.iter (fun ex ->
+                        logger.LogError (ex, "Error while processing request that generated response with documentId '{documentId}'", documentId))
+                else
+                    logger.LogWarning (
+                        ("Produced request error GraphQL response with:\n"
+                         + "- documentId: '{documentId}'\n"
+                         + "- error(s):\n    {requestError}\n"
+                         + "- metadata:\n    {metadata}\n"),
+                        documentId,
+                        errs,
+                        metadata
+                    )
```

Comment:
Maybe we just iter over without allocating new lists?
```suggestion
                let notHasException =
                    errs
                    |> Seq.map (fun err ->
                        err.Exception |> ValueOption.iter (fun ex ->
                            logger.LogError (ex, "Error while processing request that generated response with documentId '{documentId}'", documented)
                        err.Exception.IsNone
                    |> Seq.fold (fun acc notExn -> acc && notExn) true
                if notHasException then
                    logger.LogWarning (
                        ("Produced request error GraphQL response with:\n"
                         + "- documentId: '{documentId}'\n"
                         + "- error(s):\n    {requestError}\n"
                         + "- metadata:\n    {metadata}\n"),
                        documentId,
                        errs,
                        metadata
                    )
```

Reply:
Probably fold can be replaced with `Seq.all id`

Reply:
I call this "the Tetris effect in functional programming" 😄

Reply:
Done.